CMakeLists.txt

```diff
@@ -32,18 +26,48 @@ else()
   message(FATAL_ERROR "Required dependency GMP not found. Set GMP_LIB_DIR and GMP_INCLUDE_DIR to inform cmake where to look")
 endif()
 
+# Find PYTHON_EXECTUABLE first based on $PATH
```

Comment:
I think after this gets merged in we can split out all this code into a cmake/FindPython.cmake library